src/index.ts

```diff
@@ -62,6 +62,31 @@ class IconFontBuildr {
 
     this.config = mergeWith ( {}, this.configDefault, config, ( prev, next ) => isArray ( next ) ? next : undefined );
 
+    this.config.icons = Object.keys ( this.config.icons ).reduce( ( obj, val ) => {
+        const iconConfig = this.config.icons [ val ],
+              key = iconConfig.icon || iconConfig
+
+        obj [ key ] = {
+          icon: key,
+          name: iconConfig.name || key,
+          // default values:
+          codepoints: undefined,
+          ligatures: [ key.replace ( /-/g, '_' ) ] // Hyphens aren't supported
+        };
+
+        // deduplicating user input and filtering empty entries
+        if ( iconConfig.codepoints && Array.isArray ( iconConfig.codepoints )  && iconConfig.codepoints.length ) {
+          obj [ key ].codepoints = iconConfig.codepoints
+            .filter( ( value, index, array ) => ( value && array.indexOf(value) === index ) )
```

Comment:
these sorts of functions should kind of never be written, especially here that we are using `lodash`, just calling `_.uniq` would be preferable.

Reply:
I never really used lodash and didn't know of `_.uniq` 😮
Better use that one then